microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,512 @@
+    private static final EnvironmentParameters EPOLL_POOLED_PARAMS =
+            new EpollEnvironmentParametersBase(PooledByteBufAllocator.DEFAULT);
+    private static final EnvironmentParameters OIO_UNPOOLED_PARAMS =
+            new OioEnvironmentParametersBase(UnpooledByteBufAllocator.DEFAULT);
+    private static final EnvironmentParameters OIO_POOLED_PARAMS =
+            new OioEnvironmentParametersBase(PooledByteBufAllocator.DEFAULT);
+
+    public static enum EnvironmentType {
+        EMBEDDED_POOLED, EMBEDDED_UNPOOLED,
+        NIO_POOLED, NIO_UNPOOLED,
+        EPOLL_POOLED, EPOLL_UNPOOLED,
+        OIO_POOLED, OIO_UNPOOLED;
+    }
+
+    public static enum DataPayloadType {
+        SMALL, MEDIUM, LARGE, JUMBO;
+    }
+
+    private static final Map<EnvironmentType, Environment> ENVIRONMENTS = new HashMap<EnvironmentType, Environment>();
+    private static final Map<DataPayloadType, BenchmarkTestPayload> PAYLOADS =
+            new HashMap<DataPayloadType, BenchmarkTestPayload>();
+
+    static {
+        ENVIRONMENTS.put(EnvironmentType.OIO_POOLED, boostrapEnvWithTransport(OIO_POOLED_PARAMS));
+        ENVIRONMENTS.put(EnvironmentType.OIO_UNPOOLED, boostrapEnvWithTransport(OIO_UNPOOLED_PARAMS));
+        ENVIRONMENTS.put(EnvironmentType.NIO_POOLED, boostrapEnvWithTransport(NIO_POOLED_PARAMS));
+        ENVIRONMENTS.put(EnvironmentType.NIO_UNPOOLED, boostrapEnvWithTransport(NIO_UNPOOLED_PARAMS));
+        if (Epoll.isAvailable()) {
+            ENVIRONMENTS.put(EnvironmentType.EPOLL_POOLED, boostrapEnvWithTransport(EPOLL_POOLED_PARAMS));
+            ENVIRONMENTS.put(EnvironmentType.EPOLL_UNPOOLED, boostrapEnvWithTransport(EPOLL_UNPOOLED_PARAMS));
+        }
+        ENVIRONMENTS.put(EnvironmentType.EMBEDDED_POOLED, boostrapEmbeddedEnv(PooledByteBufAllocator.DEFAULT));
+        ENVIRONMENTS.put(EnvironmentType.EMBEDDED_UNPOOLED, boostrapEmbeddedEnv(UnpooledByteBufAllocator.DEFAULT));
+        PAYLOADS.put(DataPayloadType.SMALL, createPayload(DataPayloadType.SMALL));
+        PAYLOADS.put(DataPayloadType.MEDIUM, createPayload(DataPayloadType.MEDIUM));
+        PAYLOADS.put(DataPayloadType.LARGE, createPayload(DataPayloadType.LARGE));
+        PAYLOADS.put(DataPayloadType.JUMBO, createPayload(DataPayloadType.JUMBO));
+    }
+
+    @Param
+    public EnvironmentType environmentType;
+
+    @Param
+    public DataPayloadType dataType;
+
+    @Param({ "0", "255" })
+    public int padding;
+
+    @AfterClass
+    public static void teardown() {
+        for (Environment env : ENVIRONMENTS.values()) {
+            try {
+                env.teardown();
+            } catch (Exception e) {
+                handleUnexpectedException(e);
+            }
+        }
+        for (BenchmarkTestPayload payload : PAYLOADS.values()) {
+            payload.release();
+        }
+    }
+
+    @Benchmark
+    public void writeData() {
+        Environment env = ENVIRONMENTS.get(environmentType);
+        if (env == null) {
```

Comment:
I was not aware of the `@Setup` macro.  That is a better approach though.  I will see if I can work that in.  Thanks!